custom_components/kohler/__init__.py

```diff
@@ -398,15 +429,24 @@ def _updateUniqueId(self, hass, platform, old_uid, new_uid):
 
     def updateBinarySensor(self, sensor: KohlerDataBinarySensor):
         state = None
+        if isinstance(sensor, KohlerDataOutletBinarySensor):
+            outletSensor: KohlerDataOutletBinarySensor = sensor
+            outletSensor.state = self.isOutletOn(
+                outletSensor.valve, outletSensor.outlet
+            )
+        else:
+            if sensor.systemKey:
```

Comment:
Can we provide a comment around these if statements, I'm not sure what systemKey or valueKey is.